### ISSUE TITLE
MongoDB auth error

### ISSUE THREAD

**Issue by monoton** [erlio/vernemq#857]
### Environment

- VerneMQ Version: 1.5.0
- OS: Centos 7 (Docker 18.06)
- Erlang/OTP version (if building from source): 20

### Preface
We are added some required for us attributes into document to solve autodelete document issue
```
{
	"_id": ObjectId("5ba8d21797f1f8adc3311654"),
	"client_id": "90019b20-d7a3-42df-9bdf-81670a34de20",
	"created": ISODate("2018-09-24T11:40:02.806Z"),
	"mountpoint": "",
	"passhash": "$2a$10$Ddo3V6cq5j4adJdjhDssGej4GVyjomCe6fD6RDt3vqQiCeMQfdvmW",
	"publish_acl": [{
		"pattern": "test"
	}],
	"subscribe_acl": [{
		"pattern": "test"
	}],
	"system": false,
	"username": "test",
	"expired": ISODate("2018-09-24T12:40:02.906Z")
}
```
### Expected behavior

Normal authentication

### Actual behaviour
Authentication error
```
12:48:15.326 [error] can't call function auth_on_register with args [{addr,<<"10.0.112.135">>},{port,41380},{mountpoint,<<>>},{client_id,<<"90019b20-d7a3-42df-9bdf-81670a34de20">>},{username,<<"user">>},{password,<<"password">>},{clean_session,true}] in "/opt/vernemq/share/lua/auth/mongodb.lua" due to {error,badarg}
12:48:15.326 [warning] can't authenticate client {[],<<"90019b20-d7a3-42df-9bdf-81670a34de20">>} due to error
```
lua script cat parse value of `created` attribute but its regular Mongo func
Removing this attribute solves ther problem but its not right solution for us

**Comment by larshesel** [comment]
Hi, it looks like the LUA script doesn't really care about the extra items - it looks up the document by the `client_id`, `username` and `mountpoint` - and then retrieves the `publish_acl` and `subscribe_acl` files. Are you using the stock lua script?

Are you sure it's because of the `ISODate` function? Could you experiment with removing them again and adding raw "created" and "expired" fields?

I'll have to test this out to see why this fails - I probably won't have time to look into this today, though.

**Comment by monoton** [comment]
Yes we are using stock lua scripts
> Are you sure it's because of the ISODate function?

If change value of this fields to string - everything all right.
But we need to use ISOSate func

**Comment by larshesel** [comment]
It seems the mongodb driver returns the mongodb `Date` type as an erlang timestamp tuple, which the function used to map mongodb results into lua doesn't understand returns as is - which makes lua throw a badarg error

**Comment by larshesel** [comment]
There's a fix ready in #875 - it would be great if you could test it out.

**Comment by monoton** [comment]
Yes, it works
